pkg/api/validation/validation.go

```diff
@@ -1278,6 +1291,18 @@ func ValidateSecret(secret *api.Secret) errs.ValidationErrorList {
 		}
 	case api.SecretTypeOpaque, "":
 		// no-op
+	case api.SecretTypeDockercfg:
+		dockercfgBytes, exists := secret.Data[api.DockerConfigKey]
+		if !exists {
+			allErrs = append(allErrs, errs.NewFieldRequired(fmt.Sprintf("data[%s]", api.DockerConfigKey)))
+			break
+		}
+
+		dockercfg := &credentialprovider.DockerConfig{}
+		if err := json.Unmarshal(dockercfgBytes, dockercfg); err != nil {
```

Comment:
> If it's because kubelet is going to need to parse & understand the .dockercfg file, then effectively this is part of the API and this struct definition needs to be versioned etc.

It does need to parse in order to be used, but the content of secret data is actually a format owned by docker, so if we forced it to have a version and type then it would no longer be in the correct format.



> Validating that it parses doesn't really give any assurance that it'll actually work with docker.

It is necessary, but not sufficient.  However, I don't feel strongly enough about it to force its inclusion.  If you're unconvinced, I'll remove it on Monday.